2211/2/versus_checker.rb

```diff
@@ -0,0 +1,205 @@
+  end
+
+  def battle_analysis(member, file, word_per_battle)
+    Dir.chdir('..')
+    ArrayFiller.new(file, word_per_battle)
+    word_per_battle.flatten!
+    words_number_in_battle(member, word_per_battle)
+    words_number_in_rounds(member, word_per_battle)
+  end
+end
+
+# Find Bad words for this member
+class FindBadWords
+  def initialize(name, file)
+    @word = name
+    find(file)
+  end
+
+  def find(file)
+    Dir.chdir('versus-battle')
+    IO.foreach(file) do |line|
+      temp = RussianObscenity.find(line)
+      @word[:bad_words] += temp.size
+      @word[:bad_words] += 1 if /[\*]/ =~ line
+    end
+  end
+end
+
+# Sort result array
+class ResultSorter
+  def initialize(arr, attr)
+    @words = arr
+    sort_result(attr)
+  end
+
+  def sort_result(attr)
+    @words.sort_by! { |key| key[attr] * -1 }
+  end
+end
+
+# 2nd task
+class TopWords < WordControl
+  def initialize(name, number = 30)
+    @name = name
+    @number = number
+    favourite_words
+  end
+
+  def favourite_words
+    @words = []
+    member_exist
+    result
+  end
+
+  def result
+    sorted = words_sort([])
+    out = Output.new
+    out.second_result_output(@number, sorted) unless @words.empty?
+    out.output_all_members(@name) if @words.empty?
+  end
+
+  def words_sort(arr_sort)
+    @words.each { |elem| arr_sort << elem.to_a }
+    arr_sort.flatten!(1)
+    arr_sort.sort! { |first, second| (first[1] <=> second[1]) * -1 }
+  end
+
+  def member_exist
+    Dir.foreach('versus-battle') do |file|
+      for_delete = file[/\s{1}(против|vs|VS){1}\s{1}.+\z/]
+      member_name = file.chomp(for_delete)
+      word_analysis(file, []) if @name == member_name[1..-1]
+    end
+  end
+end
+
+# For pterry output ^__^
+class Output
+  def first_result_output(member)
+    printf('%-25s ', member[:name])
+    printf('| %-2d battles ', member[:battles])
+    printf('| %-4d  total bad words ', member[:bad_words])
+    printf('| %-3.2f  bad words per battle ', member[:avr_words])
+    printf('| %-4d  words per round', member[:words_per_round])
+  end
+
+  def second_result_output(number, arr)
+    number.to_i.times do |num|
+      puts "#{arr[num][0]} - #{arr[num][1]} times"
+    end
+  end
+
+  def output_all_members(name)
+    puts "Рэпер #{name} не известен мне. Зато мне известны:"
+    all_members_names
+  end
+
+  private
+
+  def all_members_names
+    Dir.foreach('versus-battle') do |file|
+      for_delete = file[/\s{1}(против|vs|VS){1}\s{1}.+\z/]
+      puts file.chomp(for_delete) unless file.include?('.') || file.include?('..')
+    end
+  end
+end
+
+# Read params
+class Parser
+  def initialize
+    doc = %{
+    Battle analysis
+
+    Usage:
+      #{__FILE__} --top-bad-words=<top_bad_words_number>
+      #{__FILE__} --top-words=<top_words_number> --name=<name>
+      #{__FILE__} --name=<name>
+
+    Options:
+      --top-bad-words=<top_bad_words_number>
+      --top-words=<top_words_number>
+      --name=<name>
+    }
+    call_doc(doc)
```

Comment:
I don't like that we call `#call_doc` directly from the initializer. I wouldn't expect that all work is done when I just do `Parser.new`.

Instead, I would implement it something like `Parser.new.parse_and_run_task`, so all the job is done not in the initializer but in the `#parse_and_run_task` method.